src/Chapter3.hs

```diff
@@ -480,6 +514,36 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = MkCastle { castleName :: String }
+
+data Wall = Wall
+
+data ThridObject = Church | Library
+
+data House = MkHouse { housePeople :: Int }
+
+data City = MkCity
+  { castle :: Maybe Castle
+  , wall :: Maybe Wall
+  , object :: ThridObject
+  , houses :: [House]
+  }
+
+buildCastle :: String -> City -> City
+buildCastle name city = city { castle = Just MkCastle { castleName = name } }
+
+buildHouse :: Int -> City -> City
+buildHouse people city = city { houses = MkHouse people : houses city }
```

Comment:
According to our task description, the house can contain only between 1 and 4 people. But as you passing `Int`, it's possible to give any number of people in the house, even the negative one. You can restrict possible values by using a safer approach.

This approach is usually called "smart constructor". To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with your `mkHouse` function.
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Int` 🙂

Reply:
:+1: